app/assets/javascripts/mediation_table.js

```diff
@@ -0,0 +1,74 @@
+var mediationTable = (function() {
+  var defaultOptions = {
+    selector: '[data-mediate-request]'
+  };
+
+  return {
+    init: function(opts) {
+      var _this = this;
+      _this.options = $.extend(defaultOptions, opts);
+      $(document).on('ready page:load', function(){
+        _this.addToggleHoldingsBehavior();
+      });
+      return this;
+    },
+
+    options: {},
+
+    addToggleHoldingsBehavior: function() {
+      var _this = this;
+      _this.mediatableRows().each(function(){
+        _this.addToggleClick($(this));
+      });
+    },
+
+    mediatableRows: function() {
+      return $(this.options.selector);
+    },
+
+    addToggleClick: function(row) {
+      var _this = this;
+      var toggle = _this.toggleHandle(row);
+      toggle.on('click', function() {
+        if(_this.rowIsProcessed(row)) {
+          _this.toggleRow(row);
+        } else {
+          _this.addHoldings(row);
+          _this.setRowProcessed(row);
+        }
+      });
+    },
+
+    toggleRow: function(row) {
+      if ( row.hasClass('expanded') ) {
+        row.removeClass('expanded');
+        row.next('tr').hide();
+      } else {
+        row.addClass('expanded');
+        row.next('tr').show();
+      }
+    },
+
+    addHoldings: function(row) {
+      $.ajax(row.data('mediate-request')).done(function(data){
```

Comment:
I wonder if it would be better to create the `<tr>` immediately (giving you a place to stash state, display a spinner, show errors, etc) and then fill in the data later.

Also, would it help to throw a `class` on the `tr`? e.g. `<tr class="holdings">`